kin-devplatform-sdk/src/main/java/kin/devplatform/KinEcosystemInitiator.java

```diff
@@ -61,100 +67,132 @@ private KinEcosystemInitiator() {
 	/**
 	 * Uses for internal initialization after process restart
 	 */
-	public void internalInit(Context context) {
-		try {
-			init(context, null);
-		} catch (BlockchainException e) {
-			EventLoggerImpl.getInstance().send(GeneralEcosystemSdkError.create(
-				ErrorUtil.getPrintableStackTrace(e), String.valueOf(e.getCode()),
-				"KinEcosystemInitiator internalInit failed"
-			));
-			e.printStackTrace();
-		}
+	public void internalInit(Context context) { // TODO: 13/01/2019 need to test and see if everything is work when we simulate process restart.
+		init(context, null, null, false, null, null, null);
+		// TODO: 10/01/2019 1. change all UI with progress bars and stuff
+		// TODO: 10/01/2019 2. init and get last KinClient without starting the migration
+		// TODO: 10/01/2019 3. fuck it all
+
 	}
 
 	/**
 	 * Uses for external (public API) initialization and jwt login.
 	 */
-	public void externalInit(Context context, KinEnvironment environment, @NonNull SignInData signInData,
-		final KinCallback<Void> loginCallback) {
+	public void externalInit(Context context, String appId, KinEnvironment environment, @NonNull SignInData signInData,
+							 final KinCallback<Void> loginCallback, @Nullable final KinMigrationListener migrationProcessCallback) {
 		if (isInitialized && isLoggedIn) {
 			fireStartCompleted(loginCallback);
 			return;
 		}
-
-		try {
-			init(context, environment);
-		} catch (BlockchainException e) {
-			fireStartError(e, loginCallback);
-			return;
+		init(context, appId, environment, true, signInData, loginCallback, migrationProcessCallback);
```

Comment:
we still have a problem here, if `isInitialized = true` and `isLoggedIn = false` we'll call init.
`init` should happen only if `isInitialized = false` o.w. call `login`

Reply:
this is ok because in the init method itself the first thing is 
if (!isInitialized) { ...}
otherwise nothing is done